core/trino-main/src/main/java/io/trino/operator/aggregation/MaxDataSizeForStats.java

```diff
@@ -39,31 +39,25 @@ private MaxDataSizeForStats() {}
 
     @InputFunction
     @TypeParameter("T")
-    public static void input(@AggregationState NullableLongState state, @BlockPosition @SqlType("T") Block block, @BlockIndex int index)
+    public static void input(@AggregationState LongState state, @NullablePosition @BlockPosition @SqlType("T") Block block, @BlockIndex int index)
     {
         update(state, block.getEstimatedDataSizeForStats(index));
     }
 
     @CombineFunction
-    public static void combine(@AggregationState NullableLongState state, @AggregationState NullableLongState otherState)
+    public static void combine(@AggregationState LongState state, @AggregationState LongState otherState)
     {
         update(state, otherState.getValue());
     }
 
-    private static void update(NullableLongState state, long size)
+    private static void update(LongState state, long size)
     {
-        if (state.isNull()) {
-            state.setNull(false);
-            state.setValue(size);
-        }
-        else {
-            state.setValue(max(state.getValue(), size));
-        }
+        state.setValue(max(state.getValue(), size));
     }
 
     @OutputFunction(StandardTypes.BIGINT)
-    public static void output(@AggregationState NullableLongState state, BlockBuilder out)
+    public static void output(@AggregationState LongState state, BlockBuilder out)
     {
-        NullableLongState.write(BigintType.BIGINT, state, out);
```

Comment:
Does it mean we will now start counting nulls as 0 (for max data size)? Could it cause issues when we will use CBO during table scans?

Reply:
That would be an unknown Estimate. But it doesn't matter because averageRowSize is set to 0 for all nulls column in TableScanStatsRule#toSymbolStatistics regardless.

Reply:
Actually we have always been converting empty to 0 when inserting the collected stats into the metastore. So this is a no-op from metastore and CBO perspective.